### PR TITLE
Asub 8194/magic link login without pw

### PR DESCRIPTION
**Pull request by accbjt**
**Be sure to run `npm test`, `npm run lint`, and write detailed test steps before requesting reviewers**

#### Jira Ticket: [ASUB-8194](https://arcpublishing.atlassian.net/browse/ASUB-8194)

Theme Blocks for Magic Link login without password.

## Test Steps
- Checkout this branch `git checkout ASUB-8194/magic-link-login-without-pw`
- `npx fusion start -f -l`
- Create a page for Log in.
- Add the `Identity Log In - Arc Block`
- Add the `Identity Login Links - Arc Block`
- Click on the option `Show Login links` under `Login`
- Click on `Show Login without password` under `Login without password`
- Create a new page called `Request One Time` with the `Page URI` pointing to `/request-one-time-password/`.
- Add the `Identity One Time Password Request Form - Arc Block` block.

[ASUB-8194]: https://arcpublishing.atlassian.net/browse/ASUB-8194?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: blocks/identity-block/features/one-time-password/default.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+				buttonLabel={phrases.t("identity-block.ota-form-button")}
+				className={BLOCK_CLASS_NAME}
+				formErrorText={error}
+				headline={phrases.t("identity-block.ota-headline")}
+				onSubmit={({ email }) => {
+					setError(null);
+					setCaptchaError(null);
+					return Identity.requestOTALink({
+						email,
+						recaptchaToken: captchaToken,
```
Comment (edwardcho1231) [inline blocks/identity-block/features/one-time-password/default.jsx]: I'm sad that it took me 2 hours to figure out why Recaptcha is not working. :(
You need to change this to
`Identity.requestOTALink(email, captchaToken)`

File: locale/en.json [hunk 2]
```diff
@@ -173,7 +173,13 @@
...
 		"identity-block.update-email-error": "There's been an error updating your email address",
 		"identity-block.update-password-error": "There's been an error updating your password",
-		"identity-block.username": "Username"
+		"identity-block.username": "Username",
+		"identity-block.ota-button": "Log in without password",
+		"identity-block.ota-headline": "Request one time log in link",
+		"identity-block.ota-subheadline": "To sign in without a password, enter your email below and you will receive a one time log in link.",
+		"identity-block.ota-form-button": "Send log in link",
+		"identity-block.ota-success-heading": "Check Your Email",
+		"identity-block.ota-success-body": "If there's an account associated with <b>%{userEmail}<\/b>, you'll get a link in your inbox that will automatically sign you in to your account. The link will expire in 24 hours."
```
Comment (edwardcho1231) [inline locale/en.json:183]: Was there a change in the message? This seems to be a little bit different from the figma.

File: blocks/identity-block/features/one-time-password/default.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		)
+	}
+
+	return (
+		<div>
+			<HeadlinedSubmitForm
+				buttonLabel={phrases.t("identity-block.ota-form-button")}
+				className={BLOCK_CLASS_NAME}
+				formErrorText={error}
+				headline={phrases.t("identity-block.ota-headline")}
```
Comment (edwardcho1231) [inline blocks/identity-block/features/one-time-password/default.jsx:56]: The figma has placeholder for the form

File: blocks/identity-block/features/one-time-password/default.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+	if (!isInitialized) {
+		return null;
+	}
+
+	if (success) {
+		return (
+			<div className={BLOCK_CLASS_NAME}>
+				<HeadingSection>
+					<Heading>{phrases.t("identity-block.ota-success-heading")}</Heading>
```
Comment (edwardcho1231) [inline blocks/identity-block/features/one-time-password/default.jsx:40]: The font and font size are different from figma on both `Check your email title` and `Request one time log in link`

**Comment by edwardcho1231** [comment]
Screenshot 2024-04-30 at 10 02 51 PM
There seems to be duplicate error messages on the block when BE API call fails.

File: blocks/identity-block/features/one-time-password/default.jsx (hunk 1 above)
Comment (accbjt) [reply blocks/identity-block/features/one-time-password/default.jsx]: @edwardcho1231 Thanks for taking your time on this. I was actually stuck on this and me and Laura wasn't able to figure it out when we were on a call.

File: locale/en.json (hunk 2 above)
Comment (accbjt) [reply locale/en.json:183]: I copied directly from the Figma designs. The original images are different from the Figma designs.

File: blocks/identity-block/features/one-time-password/default.jsx (hunk 3 above)
Comment (accbjt) [reply blocks/identity-block/features/one-time-password/default.jsx:56]: Thanks, I'll update

File: blocks/identity-block/features/one-time-password/default.jsx (hunk 4 above)
Comment (accbjt) [reply blocks/identity-block/features/one-time-password/default.jsx:40]: Okay, I'll check.

**Comment by accbjt** [comment]
@edwardcho1231 I updated most of the issues. I want to talk to the Design team regarding the Heading font size. The font size is coming from a component we already use on other pages, so I don't want to change it unless they want to change all the other pages. It is only a 4px difference so I think it might be okay but I'll ask them to make sure.

File: locale/en.json (hunk 2 above)
Comment (edwardcho1231) [reply locale/en.json:183]: The design currently shows 
`An email with a link that will automatically sign you in to your account has been sent to user@gmail.com. The link will expire in 24 hours.`
Can you check with design on which phrase they want on this?

**Comment by edwardcho1231** [comment]
@accbjt Once you check with the design, please let me know when this is ready for another look.

**Review by edwardcho1231 — DISMISSED** [review]
Other than the statement on the `identity-block.ota-success-body` mentioned on the slack dm, it LGTM. Thanks.

**Comment by nschubach** [comment]
Do we have a sane default we could use for the "Login without password URL"?  Something like `/account/one-time/` ?

image

**Comment by nschubach** [comment]
Otherwise, I don't see any technical/themes-related issues with this PR.

**Comment by accbjt** [comment]
@nschubach I updated the default url to `/account/request-one-time/`. Does that work for you?

**Review by nschubach — APPROVED** [review]
LGTM

**Comment by accbjt** [comment]
@nschubach Do you think you can merge this in for me. I do not have admin permission to merge. It's because of the Chromatic test not passing.

**Pull request merged**